R/translate-sql-helpers.R

```diff
@@ -245,11 +245,20 @@ sql_cast <- function(type) {
 
 #' @rdname sql_variant
 #' @export
-sql_try_cast <- function(type) {
-  type <- sql(type)
-  function(x) {
-    sql_expr(try_cast(!!x %as% !!type))
-    # try_cast available in MSSQL 2012+
+sql_try_cast <- function(type1, type2 = NULL) {
```

Comment:
Rather than making this function more complicated, can you please just replace the translation for `as.integer()`, expanding the comment a little more to make it more clear why this is necessary.

Reply:
(Thanks, yes, I can add a test for `as.numeric` and add a line to `NEWS.md`)

Just to clarify regarding 'just replace the translation for `as.integer()`, would you like me to leave the `sql_try_cast` function unchanged, and instead change the definition of `as.integer` in [R/backend-mssql.R](https://github.com/tidyverse/dbplyr/blob/master/R/backend-mssql.R) from using `sql_try_cast` and, say, define the function 'in-place' within [R/backend-mssql.R](https://github.com/tidyverse/dbplyr/blob/master/R/backend-mssql.R)?

e.g.

```r
as.integer = function(x) {
  sql_expr(try_cast(try_cast(!!x %as% "NUMERIC") %as% "INT"))
}
# 'try_cast' directly to INT results in returning NA for an entire column if a single row has an invalid value
# in MSSQL, NUMERIC returns a double, rounded to an integer, and
# does not return NA for an entire column if a single row has an invalid value
```

Reply:
Yup!